src/chain_sync/chain_muxer.rs

```diff
@@ -436,9 +461,24 @@ where
                         .with_label_values(&[metrics::values::PUBSUB_BLOCK])
                         .inc();
                     // Assemble full tipset from block
-                    let tipset =
-                        Self::gossipsub_block_to_full_tipset(b, source, network.clone()).await?;
-                    (tipset, source)
+                    if let GossipBlockProcessingStrategy::GetFullTipset =
```

Comment:
nit: same here, there could be less nesting if we get rid of the `else` blocks.

Reply:
Fixed.